### DISCUSSION TITLE
English Video Tutorial for Custom Font Training.

### DISCUSSION THREAD

**Discussion by OttomanZ** [PaddlePaddle/PaddleOCR#8269] [Ideas & Features]
Hi, 
I wanted @PaddlePaddle to help create a digit recognizer model for 7 Segment Displays, there are many industrial applications for such a recognizer, and there is a tutorial in Chinese on Digital Tube Character Extraction. I am attaching the required fonts for its application. It would be nice if such a model is trained and uploaded using PaddleOCR.

Currently, I am working on an industrial pump dispenser and I want to read the text from the pump but the accuracy is not good using tesseract's implementations including 7seg, ssd, and letsgodigital trained models.

Also is there a video tutorial in English demonstrating how to train PaddleOCR on custom fonts, it would be much needed, even a link to a Chinese one would do so. If not can someone guide me to train it on the following font, I will go ahead and create an English video tutorial series explaining every step with simplicity.

[fonts-DSEG_v046.zip](https://github.com/PaddlePaddle/PaddleOCR/files/9984421/fonts-DSEG_v046.zip)

**Comment by manshulgoel** [comment]
Even I am facing the same issue, I need ←→↑↓ arrows data as well along with OCR. but no success yet.

Following this discussion, so that I shall be updated on this.

**Reply by OttomanZ** [reply]
@manshulgoel Hi, I ended up using EasyOCR's approach. It's quite intuitive because of the proper documentation.

Essentially the Dataset folder is as follows 

Folder
--/
   Image-0.jpg
   Annotations.csv

The annotations file is as follows,
imgpath, text


So, using trdg See PyPI and ocrdeg by Nvidia Labs you can create and train a robust network. 

But I think Paddle is more sophisticated and better. EasyOCR is like 2019ish.

**Reply by manshulgoel** [reply]
yeah, but the challenge is easyOCR doesn't detect all digits properly whereas paddleOCR has better accuracy while doing OCR



![image](https://user-images.githubusercontent.com/45064959/201301201-50d5da10-e7c5-40ac-80d0-2644aa5f79ba.png)

**Reply by OttomanZ** [reply]
@manshulgoel Hi, can you send a few sample images. We can perhaps train a keras classifier on your data for determining the direction of the arrows. And do OCR on the rest?

**Reply by OttomanZ** [reply]
Ok, I think I have your solution you can train a YOLOv5 model on determine the regions to extract. Then run the Keras classifier on the regions.

**Reply by manshulgoel** [reply]
![IMG-20220921-WA0011](https://user-images.githubusercontent.com/45064959/201331725-695228f2-b099-4ba5-bab9-30b6c37bc8bd.jpg)
![IMG-20220927-WA0003](https://user-images.githubusercontent.com/45064959/201331732-0b7cd0bc-0508-45b0-a669-1c4de335de85.jpg)
![IMG-20220927-WA0004](https://user-images.githubusercontent.com/45064959/201331737-fc037c45-bd96-494c-9280-f450bbca61c8.jpg)


I tried keras OCR but results were not great (didn't try via yolov5)
whereasPaddle_OCR is able to read all values except arrows.

**Reply by OttomanZ** [reply]
@manshulgoel Please try Roboflow for easily training a YOLOv5 model, RoboFlow provides the labelling tools as well as a pretty good YOLO export feature which can be used to go ahead and train a YOLOv5 Medium Model. I suggest adding the Bounding Box level and Picture Level Augmentations to extract the regions, after which use that to create a dataset for Keras to make a classifier and then your issue will be properly resolved. 😀

If you need any help contact me here or at Muneeb@muneeb.co

**Comment by Zahak-Anjum** [comment]
@manshulgoel 
Any updates on this? Did you able to recognize the arrows  ←→↑↓? @OttomanZ Can you kindly elaborate on your approach? More precisely after yolov5, keras classifier will classify the characters, instead one wants to recognize them. Isn't after having the cropped image from yolov5 the right approach would be to apply paddle OCR?
